### DISCUSSION TITLE
Combining attributes for complex ldap username?

### DISCUSSION THREAD

**Discussion by boristhefirst** [authelia/authelia#6255] [Q&A]
I'm trying to build real working Authelia.
Before to go deeper in OIDC I need to understand more about the ldap authentication.

I'm using `implementation: activedirectory` and the following settings `base_dn: DC=example,DC=inter-medial-domain-name,DC=com` , `username_attribute: sAMAccountName` and
`users_filter: (&(|({username_attribute}={input})({mail_attribute}={input}))(sAMAccountType=8053...`

Is it possible to use logon name like `domain\username` (down-level logon name) for Authelia login page in the first factor and Active Directory ldap background?
I mean something like example\john.smith where john.smith is sAMAccountName .

I've tried to modify `username_attribute: domain\sAMAccountName` (exmple\sAMAccountName) and many other combinations.

I've tried also modification of `users_filter: (&(|({username_attribute}={input})({mail_attrib...` to something like `users_filter: (&(|({domain}\{username_attribute}={input})({mail_attrib...` or `users_filter: (&(|({example}\{username_attribute}={input})({mail_attrib...`  and various similar expressions without success. May be I have to change both or to modify some other settings too.

I've seen example using interesting syntax for the attributes. I've didn't met it before using 7.37.5 Bare-Metal deployment method.

```
  attributes:
    distinguished_name: 'distinguishedName'
    username: 'uid'
    display_name: 'displayName'
    mail: 'mail'
    member_of: 'memberOf'
    group_name: 'cn'
``` 

`distinguished_name: 'distinguishedName'` is new for me. So I believe that there are other attributes not known form me.
The greatest surprise would be if there is working down-level logon name for AD ldap authentication background!

**Comment by james-d-elliott** [comment]
It's not likely possible currently what would be the use case? Also what do you mean down level login name?

**Reply by boristhefirst** [reply]
When using Active Directory for a first factor authentication background through LDAP there are different formats of the logon user names.
That is often met and required especially when the AD domain is part of more complex AD forest but it's useful even in single domain scenario.
In my case the authentication of all other systems is redirected through ldap to the AD domain and Authelia is very compatible for me to define the rights and the access for different users and groups to resources using SSO.

Of course the core of the authentication logon names in AD is the `sAMAccountName`. But in most cases the systems are requiring more detailed formats for logon username to define belonging to the AD domain. That's why the derivatives of the sAMAccountName like UPN and down-level logon name are appearing.
Down-level logon name is just a formal term used from Microsoft for this format of the logon name which is including the AD NetBIOS domain name. This format is `domain\UserName` and particularly means that logon name have to be written like `example\john.smith` where `example`  is the the NetBIOS name of the AD domain and john.smith is sAMAccountName. This is easier for typing than the UPN is.

Here both formats are described

https://learn.microsoft.com/en-us/windows/win32/secauthn/user-name-formats#down-level-logon-name
